src/index.ts

```diff
@@ -224,7 +224,10 @@ export function register (options: Options = {}): Register {
 
   // Add all files into the file hash.
   for (const fileName of config.fileNames) {
-    if (/\.d\.ts$/.test(fileName)) {
+    const regexpContent = ['.d.ts'].concat(extensions).map(e => e.replace(/\./g, '\\.')).join('|')
+    // going to look something like /(\.d\.ts|\.ts|\.tsx)$/
+    const regexp = new RegExp('(' + regexpContent + ')$')
+    if (regexp.test(fileName)) {
```

Comment:
Let’s include everything without using a RegExp if we are doing this. Using only `.d.ts` before was because TypeScript will give us a list of the entire project (e.g. when using includes) while when running the CLI we may use none of the files anyway.

Reply:
Makes sense